storages/backends/gcloud.py

```diff
@@ -141,12 +144,22 @@ def get_default_settings(self):
             # roll over.
             "max_memory_size": setting("GS_MAX_MEMORY_SIZE", 0),
             "blob_chunk_size": setting("GS_BLOB_CHUNK_SIZE"),
+            # use in cases where service account key isn't available in env
+            # in such cases, sign blob api is REQUIRED for signing data
+            "iam_sign_blob": setting("GS_IAM_SIGN_BLOB", False),
+            "sa_email": setting("GS_SA_EMAIL"),
         }
 
     @property
     def client(self):
         if self._client is None:
+            if self.project_id is None or self.credentials is None:
+                self.credentials, self.project_id = auth.default(
+                    scopes=['https://www.googleapis.com/auth/cloud-platform']
+                )
             self._client = Client(project=self.project_id, credentials=self.credentials)
+        if self.credentials and self.credentials.token_state != TokenState.FRESH:
+            self.credentials.refresh(requests.Request())
```

Comment:
Have a couple questions I'm hoping you can expand upon:

1. Why don't we guard the `auth.default` behind `iam_sign_blob` rather than project_id & credentials?
2. Should it be an `and` check on project_id & credentials?
3. Similar question on guarding `credentials.refresh`

Reply:
Yeah I have reverted the changes on the Client property and made the auth default use under the iam_sign_blob condition